differ/storage.go

```diff
@@ -537,18 +537,24 @@ func (storage DBStorage) ReadLastNotifiedRecordForClusterList(
 		clusterIDs[idx] = string(entry.ClusterName)
 	}
 
+	selectQuery := `SELECT org_id, cluster, report, notified_at FROM ( SELECT DISTINCT ON (cluster) * FROM reported`
 	whereClause := fmt.Sprintf(` WHERE event_type_id = %v AND state = 1 AND org_id IN (%v) AND cluster IN (%v)`,
 		eventTarget, inClauseFromStringSlice(orgIDs), inClauseFromStringSlice(clusterIDs))
+	orderByClause := `ORDER BY cluster, notified_at DESC) t`
+	var err error
+	var rows *sql.Rows
+
+	if len(timeOffset) == 0 || strings.Split(timeOffset, " ")[0] == "0" {
```

Comment:
this one might panic

Reply:
How? The `Split` is not executed in case of empty `timeOffset`

Reply:
" ", "   ", "    " perhaps?

Reply:
there's no sanitization of this value before IIRC

Reply:
let me check. I assumed it is sanitised when read from the toml, but maybe that's wrong assumption.

Reply:
nope, not sanitised at all!
fixing